src/simple_lottery/init.rs

```diff
@@ -30,9 +30,16 @@ pub fn make_setup(params: &Params) -> Setup {
         if (middle <= left) || (middle >= right) {
             let u = bound_soundness.max(bound_completeness).ceil();
             let mu = u * ratio_completeness;
-            return Setup {
-                proof_size: u as u64,
-                lottery_probability: mu / params.set_size as f64,
+            return if params.lower_bound < u as u64 {
+                Setup {
+                    proof_size: params.lower_bound,
+                    lottery_probability: params.lower_bound as f64 / params.set_size as f64,
+                }
```

Comment:
I think this should be
```suggestion
                Setup {
                    proof_size: params.lower_bound + 1,
                    lottery_probability: 1.0,
                }
```

Reply:
@tolikzinovyev the lottery probability would be one only if `nf=np` no ?

It still needs to be corrected to `(params.lower_bound as f64 + 1.0) / params.set_size as f64`

Reply:
The analysis from the paper doesn't work here anymore, we need a different approach. We are proving knowledge of n_f + 1 elements. By requiring n_f + 1 elements in the proof, we have soundness. Since we set p = 1, we have completeness.

With your current code, we certainly don't have completeness. The honest prover will on average have n_f elements passing the lottery, but with probability ≈1/2 it will be less.

Reply:
Right I forgot about it Shouldn't we set the lottery_probability to `1-2^-lambda_completeness, nf/np` then?

Reply:
Why? 1 works great already.

Reply:
(or a function of `lambda_completeness`?

Reply:
this would artificially increase the completeness error, there is no reason to do it

Reply:
Done